packages/devtools_app/lib/src/screens/memory/allocation_profile_table_view.dart

```diff
@@ -0,0 +1,424 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../../devtools_app.dart';
+import '../../config_specific/import_export/import_export.dart';
+import '../../shared/table.dart';
+import '../../shared/table_data.dart';
+
+import '../vm_developer/vm_service_private_extensions.dart';
+import 'allocation_profile_table_view_controller.dart';
+import 'panes/control/constants.dart';
+
+/// The default width for columns containing *mostly* numeric data (e.g.,
+/// instances, memory).
+const _defaultNumberFieldWidth = 80.0;
+
+class FieldClassName extends ColumnData<ClassHeapStats> {
+  FieldClassName()
+      : super(
+          'Class',
+          fixedWidthPx: scaleByFontFactor(200),
+        );
+
+  @override
+  String? getValue(ClassHeapStats dataObject) => dataObject.classRef!.name;
+
+  @override
+  String getDisplayValue(ClassHeapStats dataObject) =>
+      '${getValue(dataObject)}';
+
+  @override
+  bool get supportsSorting => true;
+}
+
+enum HeapGeneration {
+  newSpace,
+  oldSpace,
+  total;
+
+  @override
+  String toString() {
+    switch (this) {
+      case HeapGeneration.newSpace:
+        return 'New Space';
+      case HeapGeneration.oldSpace:
+        return 'Old Space';
+      case HeapGeneration.total:
+        return 'Total';
+    }
+  }
+}
+
+class FieldInstanceCountColumn extends ColumnData<ClassHeapStats> {
+  FieldInstanceCountColumn({required this.heap})
+      : super(
+          'Instances',
+          titleTooltip: 'The number of instances of the class in the heap',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  final HeapGeneration heap;
+
+  @override
+  dynamic getValue(ClassHeapStats dataObject) {
+    switch (heap) {
+      case HeapGeneration.newSpace:
+        return dataObject.newSpace.count;
+      case HeapGeneration.oldSpace:
+        return dataObject.oldSpace.count;
+      case HeapGeneration.total:
+        return dataObject.instancesCurrent;
+    }
+  }
+
+  @override
+  String getDisplayValue(ClassHeapStats dataObject) =>
+      '${getValue(dataObject)}';
+
+  @override
+  bool get numeric => true;
+
+  @override
+  bool get supportsSorting => true;
+}
+
+class FieldExternalSizeColumn extends FieldSizeColumn {
+  FieldExternalSizeColumn({required super.heap})
+      : super._(
+          title: 'External',
```

Comment:
Can you create file panes/profile/PROFILE.md (like LEAK_TRACKING.md), link it with button '(?)' like on 'Leaks' pane and put explanation here for some concepts that may be not clear for users. Like: what is external memory on different devices? 

Then we will need to review the doc with UX people (Mindy and Tao) and with writing specialist (Shams).

Reply:
The label 'This document is under construction' on the top will excuse the drafty quality.